test/SurveyDesign.jl

```diff
@@ -14,6 +14,25 @@
     # TODO: needs change; this works but isn't what the user is expecting
     srs_prob = SimpleRandomSample(apisrs; probs = fill(0.3, size(apisrs_original, 1)))
     @test srs_prob.data.probs[1] == 0.3
+    @test srs_prob.data.weights == 1 ./ srs_prob.data.probs
+
+    srs = SimpleRandomSample(apisrs, ignorefpc = true, weights = :pw )
+    @test srs.data.probs == 1 ./ srs.data.weights
+    
+    srs = SimpleRandomSample(apisrs, ignorefpc = false, weights = :pw)
+    @test srs.data.probs == 1 ./ srs.data.weights
+
+    srs_weights = SimpleRandomSample(apisrs, ignorefpc = false, weights = :fpc)
+    @test_throws srs_weights = SimpleRandomSample(apisrs, ignorefpc = false, weights = :stype)
```

Comment:
The `@test_throws` macro needs another argument, which is the error type. See the [documentation](https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test_throws). In this case the test throws a `MethodError` because of a lateral error (not because `weights` cannot be a string vector, but because the operations done on `weights` give errors for strings), which is fine for now but eventually it should throw a more specific error.